### PR TITLE
fix: Daemon step updated 'pod delete' while pod is running

### PR DESCRIPTION
**Pull request by sarabala1979**
Fixes #TODO

### CODE REVIEW

File: workflow/controller/pod_cleanup.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func (woc *wfOperationCtx) queuePodsForCleanup() {
 	for _, obj := range objs {
 		pod := obj.(*apiv1.Pod)
 		nodeID := woc.nodeID(pod)
-		if !woc.execWf.Status.Nodes[nodeID].Fulfilled() {
+		if !woc.execWf.Status.Nodes[nodeID].Phase.Fulfilled() {
```
Comment (alexec) [inline workflow/controller/pod_cleanup.go:21]: not just Completed()?

Comment (sarabala1979) [reply workflow/controller/pod_cleanup.go:21]: Fulfilled will take care skipped and Omitted nodes also.  Fulfilled is right one
```go
// Fulfilled returns whether a phase is fulfilled, i.e. it completed execution or was skipped or omitted
func (phase NodePhase) Fulfilled() bool {
	return phase.Completed() || phase == NodeSkipped || phase == NodeOmitted
}
```

**Comment by sarabala1979** [comment]
@alexec can you review this PR? once it is approved and merged I will start v3.3.2

File: workflow/controller/pod_cleanup.go (hunk 1 above)
Comment (alexec) [reply workflow/controller/pod_cleanup.go:21]: strange they are inconsistent

**Review by alexec — APPROVED** [review]

**Pull request merged**